### ISSUE TITLE
Particles are lower than the mesh

### ISSUE THREAD

**Issue by PavelBlend** [taichi-dev/taichi_elements#44]
Here is a screenshot:
![01](https://user-images.githubusercontent.com/7983249/79636484-a3ac5a80-8180-11ea-98d4-49214685c78e.jpg)

There is no gravity in this scene.

This error is reproduced in this scene:
[test_single_voxel.zip](https://github.com/taichi-dev/taichi_elements/files/4496537/test_single_voxel.zip)

And at this commit:
d1848fdecd1318e16e31dcf6ba70921bec1f574f

**Comment by yuanming-hu** [comment]
This is probably hard to make perfect given the current rasterizer. Since each voxel is either filled or not, there is always some quantization errors. If we use a higher resolution, however, the errors will be much less pronounced.

**Comment by PavelBlend** [comment]
I had such an assumption:

maybe there is no quantization error? It seems to me that all particles are simply shifted one voxel down. Because if all the particles are raised one voxel up, then everything will be right. I created an emitter exactly along the borders of the voxels and I got this result:
![01](https://user-images.githubusercontent.com/7983249/79822224-52c37e80-8399-11ea-815e-bc71f80fa7cf.jpg)
